src/elements/content-sidebar/MetadataSidebarRedesign.tsx

```diff
@@ -70,25 +71,43 @@ function MetadataSidebarRedesign({
     const { formatMessage } = useIntl();
 
     const [selectedTemplates, setSelectedTemplates] = React.useState<Array<MetadataTemplate>>([]);
-    const [editingTemplate, setEditingTemplate] = React.useState<MetadataTemplateInstance | MetadataTemplate | null>(
-        null,
-    );
+    const [editingTemplate, setEditingTemplate] = React.useState<MetadataTemplateInstance | null>(null);
     const [isUnsavedChangesModalOpen, setIsUnsavedChangesModalOpen] = React.useState<boolean>(false);
 
-    const { file, templates, errorMessage, status, templateInstances } = useSidebarMetadataFetcher(
-        api,
-        fileId,
-        onError,
-        isFeatureEnabled,
-    );
+    const { createMetadataInstance, file, templates, errorMessage, status, templateInstances, updateMetadataInstance } =
+        useSidebarMetadataFetcher(api, fileId, onError, isFeatureEnabled);
 
     const handleUnsavedChanges = () => {
         setIsUnsavedChangesModalOpen(true);
     };
 
+    const convertToInstance = (template: MetadataTemplate): MetadataTemplateInstance => {
+        return {
+            canEdit: !!file.permissions.can_upload,
+            displayName: template.displayName,
+            hidden: template.hidden,
+            id: template.id,
+            fields: template.fields,
+            scope: template.scope,
+            templateKey: template.templateKey,
+            type: template.type,
+        };
+    };
+
     const handleTemplateSelect = (selectedTemplate: MetadataTemplate) => {
         setSelectedTemplates([...selectedTemplates, selectedTemplate]);
-        setEditingTemplate(selectedTemplate);
+        setEditingTemplate(convertToInstance(selectedTemplate));
+    };
+
+    const checkIfTemplateInstanceExists = (id: string): MetadataTemplateInstance => {
+        return templateInstances.find(templateInstance => templateInstance.id === id);
+    };
+
+    const handleSubmitInstance = (values: FormValues) => {
+        const existingTemplateInstance = checkIfTemplateInstanceExists(editingTemplate?.id);
```

Comment:
is it necessary for this function to return Instance? Looks like boolean should be good enough